src/main/java/mate/academy/carsharing/service/UserServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+package mate.academy.carsharing.service;
+
+import jakarta.transaction.Transactional;
+import java.util.Optional;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import mate.academy.carsharing.dto.UpdateUserProfileRequestDto;
+import mate.academy.carsharing.dto.UpdateUserRoleRequestDto;
+import mate.academy.carsharing.dto.UserRequestDto;
+import mate.academy.carsharing.dto.UserResponseDto;
+import mate.academy.carsharing.exception.EntityNotFoundException;
+import mate.academy.carsharing.exception.RegistrationException;
+import mate.academy.carsharing.mapper.UserMapper;
+import mate.academy.carsharing.model.Role;
+import mate.academy.carsharing.model.User;
+import mate.academy.carsharing.repository.RoleRepository;
+import mate.academy.carsharing.repository.UserRepository;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Transactional
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+    private final PasswordEncoder passwordEncoder;
+    private final RoleRepository roleRepository;
+
+    @Override
+    public UserResponseDto register(UserRequestDto userRequestDto) throws RegistrationException {
+        if (userRepository.existsByEmail(userRequestDto.getEmail())) {
+            throw new RegistrationException("Such email already exists");
```

Comment:
provide parameter to message
```suggestion
            throw new RegistrationException("User with this email already exists");
```